tests/test_cli.py

```diff
@@ -139,3 +189,62 @@ def test_triton_profile(self, mocker, monkeypatch):
         args = parse_args()
         args.func(args)
         mock_run.assert_called_once_with(["genai-perf", "-m", "add_sub"], check=True)
+
+    @pytest.mark.parametrize("model", ["mock_llm"])
+    def test_triton_metrics(self, model, setup_and_teardown):
+        # Import the Model Repo
+        pid = utils.run_server(repo=MODEL_REPO)
+        setup_and_teardown.pid = pid
+        utils.wait_for_server_ready()
+
+        # Inference the Model
+        self._infer(model, prompt=PROMPT)
+
+        output = ""
+        # Redirect stdout to a buffer to capture the output of the command.
+        with io.StringIO() as buf, redirect_stdout(buf):
```

Comment:
What do you think of adding some helpers/refactors around redirecting stdout and also moving json parsing to the helpers, something like:
```python
def run_and_capture_stdout(args):
    """
    Runs Triton CLI command, captures the stdout, and returns it for parsing.
    """
    with io.StringIO() as buf, redirect_stdout(buf):
        run(args)
    return buf.getvalue()

def _metrics(args):
    args = ["metrics"]
    output = run_and_capture(args)
    return json.loads(output)

...

def test_triton_metrics(...):
    metrics = self._metrics()
    
    for loaded_models in metrics[...]:
        ...
```

Reply:
@rmccorm4 I really like the idea of standardization introduced by `run_and_capture_stdout(args)` and have added it! 

But would wrapping `json.loads()` in the helper functions make debugging harder? For example, let us assume `triton status` erroneously returns an empty output. Then an error would be thrown by the helper function rather than the error being thrown from directly within the `test_triton_metrics()` function.

Reply:
It would be thrown in the helper function, but should bubble up to the test that called it. Not super opinionated either way right now, but for testing purposes I think it makes sense to have the helper function do the `json.loads()` for now instead of adding an extra line or two to any test that calls it. Can always change it later if we find it causing more harm than good.